arches/app/models/resource.py

```diff
@@ -70,14 +73,39 @@ def __init__(self, *args, **kwargs):
         # end from models.ResourceInstance
         self.tiles = []
         self.descriptor_function = None
-        try:
-            self.serialized_graph = (
-                models.PublishedGraph.objects.filter(publication=self.graph.publication.publicationid, language=settings.LANGUAGE_CODE)
-                .first()
-                .serialized_graph
-            )
-        except AttributeError:
-            self.serialized_graph = None
+
+    def get_serialized_graph(self):
+        if not self.serialized_graph:
+            try:
+                self.serialized_graph = (
+                    models.PublishedGraph.objects.filter(publication=self.graph.publication.publicationid, language=settings.LANGUAGE_CODE)
+                    .first()
+                    .serialized_graph
+                )
+            except AttributeError:
+                self.serialized_graph = None
+        return self.serialized_graph
+
+    def set_serialized_graph(self, serialized_graph):
+        self.serialized_graph = serialized_graph
+
+    def get_node_datatypes(self):
+        if not self.node_datatypes:
+            self.node_datatypes = {str(nodeid): datatype for nodeid, datatype in models.Node.objects.values_list("nodeid", "datatype")}
+        return self.node_datatypes
+
+    def set_node_datatypes(self, node_datatypes):
+        self.node_datatypes = node_datatypes
+
+    def get_root_ontology(self):
+        """
+        Finds and returns the ontology class of the instance's root node
+
+        """
+        if "topnode" in self.serialized_graph:
```

Comment:
@bferguso I found a case where the serialized graph wasn't available when get_root_ontology was called. This would fix it:

`if "topnode" in self.get_serialized_graph():`